recipes/imagemagick/all/conanfile.py

```diff
@@ -100,336 +145,259 @@ def configure(self):
-            self.requires("openjpeg/2.4.0")
+            self.requires("openjpeg/2.5.0")
         if self.options.with_pango:
-            self.requires("pango/1.50.7")
+            self.requires("pango/1.50.10")
         if self.options.with_png:
-            self.requires("libpng/1.6.37")
+            self.requires("libpng/1.6.38")
         if self.options.with_tiff:
-            self.requires("libtiff/4.3.0")
+            self.requires("libtiff/4.4.0")
         if self.options.with_webp:
-            self.requires("libwebp/1.2.0")
+            self.requires("libwebp/1.2.4")
         if self.options.with_xml2:
-            self.requires("libxml2/2.9.10")
+            self.requires("libxml2/2.9.14")
         if self.options.with_freetype:
-            self.requires("freetype/2.10.4")
-        if self.options.with_djvu:
-            # FIXME: missing djvu recipe
-            self.output.warn(
-                "There is no djvu package available on Conan (yet). This recipe will use the one present on the system (if available)."
-            )
+            self.requires("freetype/2.12.1")
+        if self.options.with_zstd:
+            self.requires("zstd/1.5.2")
+        if self.options.with_fftw:
+            self.requires("fftw/3.3.9")
 
     def source(self):
-        tools.get(
-            **self.conan_data["sources"][self.version]["source"],
-            destination=self._source_subfolder,
-            strip_root=True
-        )
-
-        if self._is_msvc:
-            visualmagick_version = list(
-                self.conan_data["sources"][self.version]["visualmagick"].keys()
-            )[0]
-            tools.get(
-                **self.conan_data["sources"][self.version]["visualmagick"][
-                    visualmagick_version
-                ],
-                destination="VisualMagick",
-                strip_root=True
-            )
+        files.get(self,
+                  **self.conan_data["sources"][self.version],
+                  destination=self.source_folder,
+                  strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        env = tc.environment()
+
+        if microsoft.is_msvc(self):
+            # FIXME: it seems that PKG_CONFIG_PATH is added as a unix style path
+            env.unset("PKG_CONFIG_PATH")
+            env.define("PKG_CONFIG_PATH", self.generators_folder)
+
+            # FIXME: otherwise configure reports "ld" as its linker
+            env.define("LD", "link")
+            # use "compile" script as a compiler driver for cl, since autotools
+            # doesn't work well with cl
+            compile_script = microsoft.unix_path(
+                self, os.path.join(self.source_folder, "config", "compile"))
+            env.define("CC", f'{compile_script} cl.exe -nologo')
+            env.define("CXX", f"{compile_script} cl.exe -nologo")
+
+            env.append("CXXFLAGS", "-FS")
+            env.append("CFLAGS", "-FS")
```

Comment:
Can these be set with the toolchain?

```suggestion
            tc.extra_cxxflags.append("-FS")
            tc.extra_cflags.append("-FS")
```